projects/adrv9001/src/hal/no_os_platform.c

```diff
@@ -0,0 +1,671 @@
+*******************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include "adi_platform.h"
+#include "parameters.h"
+#include "gpio.h"
+#include "gpio_extra.h"
+#include "spi.h"
+#include "spi_extra.h"
+#include "error.h"
+#include "delay.h"
+#include "adi_common_error.h"
+#include "Navassa_EvaluationFw.h"
+#include "ORxGainTable.h"
+#include "RxGainTable.h"
+#include "TxAttenTable.h"
+
+/**
+ * @brief Opens all neccessary files and device drivers for a specific device
+ *
+ * @param devHalCfg Pointer to device instance specific platform settings
+ *
+ * @retval ADI_HAL_OK Function completed successfully, no action required
+ * @retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ * @retval errors returned by other function calls.
+ */
+int32_t no_os_HwOpen(void *devHalCfg)
+{
+	int32_t ret;
+	adi_hal_Cfg_t *phal = (adi_hal_Cfg_t *)devHalCfg;
+	struct gpio_init_param gip_gpio_reset;
+#if defined(ADRV9002_RX2TX2)
+	struct gpio_init_param gip_gpio_ssi_sync;
+#endif
+	struct xil_gpio_init_param gip_extra = {
+#ifdef PLATFORM_MB
+		.type = GPIO_PL,
+#else
+		.type = GPIO_PS,
+#endif
+		.device_id = GPIO_DEVICE_ID
+	};
+	struct xil_spi_init_param sip_extra = {
+#ifdef PLATFORM_MB
+		.type = SPI_PL,
+#else
+		.type = SPI_PS,
+#endif
+		.device_id = SPI_DEVICE_ID,
+		.flags = 0
+	};
+
+	/* Reset GPIO configuration */
+	gip_gpio_reset.number = GPIO_RESET;
+	gip_gpio_reset.extra = &gip_extra;
+	gip_gpio_reset.platform_ops = &xil_gpio_platform_ops;
+	ret = gpio_get(&phal->gpio_reset_n, &gip_gpio_reset);
+	if (ret)
+		return ret;
+
+	ret = gpio_direction_output(phal->gpio_reset_n, GPIO_HIGH);
+	if (ret)
+		return ret;
+
+#if defined(ADRV9002_RX2TX2)
+	/* SSI Sync GPIO configuration */
+	gip_gpio_ssi_sync.number = GPIO_SSI_SYNC;
+	gip_gpio_ssi_sync.extra = &gip_extra;
+	gip_gpio_ssi_sync.platform_ops = &xil_gpio_platform_ops;
+	ret = gpio_get(&phal->gpio_ssi_sync, &gip_gpio_ssi_sync);
+	if (ret < 0)
+		return ret;
+
+	ret = gpio_direction_output(phal->gpio_ssi_sync, GPIO_LOW);
+	if (ret)
+		return ret;
+#endif
+	struct spi_init_param sip = {
+		.max_speed_hz = 20000000,
+		.mode = SPI_MODE_0,
+		.chip_select = SPI_CS,
+		.platform_ops = &xil_platform_ops,
+		.extra = &sip_extra
+	};
+	ret = spi_init(&phal->spi, &sip);
+	if (ret)
+		return ret;
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * @brief Gracefully shuts down the the hardware closing any open resources
+ *        such as log files, I2C, SPI, GPIO drivers, timer resources, etc.
+ *
+ * @param devHalCfg Pointer to device instance specific platform settings
+ *
+ * @retval ADI_HAL_OK Function completed successfully, no action required
+ * @retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ */
+int32_t no_os_HwClose(void *devHalCfg)
+{
+	int32_t ret;
+	adi_hal_Cfg_t *phal = (adi_hal_Cfg_t *)devHalCfg;
+	ret = gpio_remove(phal->gpio_reset_n);
+	if (ret)
+		return ret;
+
```

Comment:
If `ADRV9002_RX2TX2` should the `gpio_ssi_sync` GPIO be freed as well?

Reply:
yes, good point